integration/rootless/rootless_test.sh

```diff
@@ -27,19 +27,39 @@ setup() {
 
 cleanup() {
 	sudo sed -i -e 's/^.*\(rootless\)/# \1/g' /usr/share/defaults/kata-containers/configuration.toml
-	sudo crictl stopp "$pod_id" || true
-	sudo crictl rmp "$pod_id" || true
+	sudo crictl stopp "$pod_id" &>/dev/null || true
+	sudo crictl rmp "$pod_id" &>/dev/null || true
 }
 
 run() {
 	pod_id="$(sudo crictl runp -r kata "${dir_path}/rootless-pod.json")"
 	waitForProcess 15 3 "sudo crictl inspectp "$pod_id" | jq '.status.state' | grep 'SANDBOX_READY'"
+
+	# There must be created one and only one temporary user.
+	local grep_uid="grep -e '^kata-[0-9]\+' /etc/passwd"
+	[ "$(eval $grep_uid | wc -l)" -eq 1 ] || \
+		die "Unexpected more than one kata temporary UID"
+	local kata_uid="$(eval $grep_uid | cut -d: -f3)"
+	local kata_gid="$(eval $grep_uid | cut -d: -f4)"
+	[[ -n "$kata_uid" && -n "$kata_gid" ]] || \
+		die "Unabled to find the kata temporary UID"
+
+	# Assert the QEMU process is owned by that user.
+	ps -u "$kata_uid" | grep "qemu" &>/dev/null || \
```

Comment:
Is it possible that some other unrelated process gets caught by this `grep "qemu"`  ? What about extracting the name of the QEMU executable out of `configuration.toml` and passing its `basename` to something like `ps -C ${qemu_binary} -o uid --no-headers` ?

Reply:
Isn't that the case that if there is more than one process, it's likely to be another qemu instance? How would your ps help? IOW, what are you trying to filter out? A "qemu-img" process for example?

Reply:
Ah, unless you were thinking about the frequent case where a `ps | grep` finds the `grep` process. But then that cannot happen here because the `grep` does not run as user `$kata_uid`.

Reply:
> Is it possible that some other unrelated process gets caught by this `grep "qemu"` ? What about extracting the name of the QEMU executable out of `configuration.toml` and passing its `basename` to something like `ps -C ${qemu_binary} -o uid --no-headers` ?

The way I understood rootless, kata is going to create a temporary user (kata-`RANDOM_ID`) for every and each sandbox. The QEMU process and some control files should be owned by that user. Here I list all the processes owned by the that temporary user, and checking there is a match to "qemu". It is hard to tell exactly what is the binary name because the build scripts allow users to overwrite it; for example, there are jobs using "qemu-tdx". An alternative is to get the path to the hypervisor set on kata's config file.

Do you think it should be more accurate?

Reply:
ah, there is also another approach: call the `check_processes()` on setup and fail if any QEMU process is actually running at test time. It should work fine with CI but it mightl make developers unhappy when they are running the tests locally.

Reply:
> > Is it possible that some other unrelated process gets caught by this `grep "qemu"` ? What about extracting the name of the QEMU executable out of `configuration.toml` and passing its `basename` to something like `ps -C ${qemu_binary} -o uid --no-headers` ?
> 
> The way I understood rootless, kata is going to create a temporary user (kata-`RANDOM_ID`) for every and each sandbox. The QEMU process and some control files should be owned by that user. Here I list all the processes owned by the that temporary user, and checking there is a match to "qemu". It is hard to tell exactly what is the binary name because the build scripts allow users to overwrite it; for example, there are jobs using "qemu-tdx". An alternative is to get the path to the hypervisor set on kata's config file.

My suggestion is precisely to get the path to QEMU from the config file :smiley:

> 
> Do you think it should be more accurate?

Yes.

Reply:
@gkurz ok, what do you think now?